restapi/src/main/java/io/stargate/web/resources/v2/schemas/IndexesResource.java

```diff
@@ -0,0 +1,262 @@
+      notes = "Drop an index",
+      response = SuccessResponse.class,
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = SuccessResponse.class),
+        @ApiResponse(code = 400, message = "Bad request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 409, message = "Conflict", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  @Path("/{indexName}")
+  public Response dropIndex(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "Name of the table to use for the request.", required = true)
+          @PathParam("indexName")
+          final String indexName,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
+
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        String.format("Keyspace '%s' not found.", keyspaceName),
+                        Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          Optional<Table> table =
+              keyspace.tables().stream()
+                  .filter(t -> t.indexes().stream().anyMatch(i -> indexName.equals(i.name())))
+                  .findFirst();
+
+          if (!table.isPresent()) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        String.format("Index '%s' not found.", indexName),
+                        Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          dropIndex(keyspaceName, indexName, authenticatedDB, table);
+
+          return Response.status(Response.Status.NO_CONTENT).build();
+        });
+  }
+
+  private void dropIndex(
+      String keyspaceName, String indexName, AuthenticatedDB authenticatedDB, Optional<Table> table)
+      throws UnauthorizedException, InterruptedException, java.util.concurrent.ExecutionException {
+    db.getAuthorizationService()
```

Comment:
Same as above, we should authorize before checking things.

Reply:
I have refactored. Please, see if this is right.